src/compiler/gcc.rs

```diff
@@ -341,6 +354,20 @@ where
             | Some(Arch(_))
             | Some(PassThrough(_))
             | Some(PassThroughPath(_)) => &mut common_args,
+            Some(ProfileUse(path)) => {
+                debug_assert!(compiler_kind == CCompilerKind::Clang);
+
+                let mut path = cwd.join(path);
+
+                // Clang allows specifying a directory here, in which case it
+                // will look for the file `default.profdata` in that directory.
+                if path.is_dir() {
+                    path.push("default.profdata");
+                }
+
+                extra_hash_files.push(path);
```

Comment:
This feels like it belongs more in the `match` above? I'm not sure why `ExtraHashFile` has logic down here, this match exists just to split arguments up into ones that get passed to the preprocessor or not.